dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorLKMM.java

```diff
@@ -206,15 +207,18 @@ public List<Event> visitLKMMXchg(LKMMXchg e) {
 
     @Override
     public List<Event> visitLKMMLock(LKMMLock e) {
-        Register dummy = e.getFunction().newRegister(e.getAccessType());
-        Expression nonzeroDummy = expressions.makeBooleanCast(dummy);
+        Type type = types.getBooleanType();
+        Register dummy = e.getFunction().newRegister(type);
```

Comment:
Same as for the other locking primitives: you load a bool but store integers which is dangerous.